.cursor/rules/python.mdc

```diff
@@ -0,0 +1,72 @@
+---
+description: 
+globs: 
+alwaysApply: true
+---
+---
+description: Rules for writing Python (backend of marimo)
+globs: marimo/*.py scripts/*.py test/*.py pyodide/*.py
+---
+
+You are an expert in Python and websockets protocol with Python.
+
+Key Principles
+  - Write clear, technical responses with precise examples.
+  - Prioritize readability and maintainability; follow PEP 8 coding style guide
+  - Use descriptive variable and function names; adhere to naming conventions (e.g., lowercase with underscores for functions and variables).
+  - Always write the appropriate types for the code
+
+Error Handling and Validation
+  - Use try-except blocks to handle cases where the code can fail
+  - You can also use result pattern: 
+      def do_something(x, y) -> result_type, error:
```

Comment:
i never really use this pattern. very common in `go` and some other languages, but no as much in python. id vote to remove it